src/components/FilterPresetsSelect/FilterPresetsSelect.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Box,
+  Dropdown,
+  DropdownButton,
+  List,
+  sprinkles,
+  Text,
+  vars,
+} from "@saleor/macaw-ui/next";
+import React, { MouseEvent } from "react";
+import { useIntl } from "react-intl";
+
+import { FilterPresetItem } from "./FilterPresetItem";
+import { messages } from "./messages";
+
+interface FilterPresetsSelectProps {
+  activePreset: number;
+  savedPresets: string[];
+  selectAllLabel: string;
+  isCustomPreset: boolean;
+  onSelectAll: () => void;
+  onRemove: (filterIndex: number) => void;
+  onSelect: (filterIndex: number) => void;
+}
+
+export const FilterPresetsSelect = ({
+  onSelect,
+  onRemove,
+  savedPresets,
+  activePreset,
+  onSelectAll,
+  selectAllLabel,
+  isCustomPreset,
+}: FilterPresetsSelectProps) => {
+  const intl = useIntl();
+
+  const getLabel = () => {
+    if (isCustomPreset) {
+      return `(${intl.formatMessage(messages.unsavedPreset)})`;
+    }
+
+    if (activePreset === 0) {
+      return selectAllLabel;
+    }
+
+    if (activePreset) {
+      return savedPresets[activePreset - 1];
+    }
+
+    return "";
+  };
+
+  const handleSelectPreset = (e: MouseEvent<HTMLElement>, index: number) => {
+    const target = e.target as HTMLElement;
+    // Only allow selecting the preset if the user clicks on the list item or the span
+    if (!["LI", "SPAN"].includes(target.tagName)) {
+      return;
+    }
+
+    onSelect(index);
+  };
+
+  return (
+    <Box display="flex" alignItems="center">
+      <Text
+        variant="caption"
+        className={sprinkles({
```

Comment:
**suggestion**: maybe it will be worth adding the ability to use whole sprinkles in MacawUI? If so I can create a task for that.

Reply:
It would be nice to jjust provide margin props as for Box

Reply:
I think from `0.8.0-64` you can add margins to `Text`